source/filter.js

```diff
@@ -21,7 +21,10 @@ import keys from './keys';
  * @memberOf R
  * @since v0.1.0
  * @category List
- * @sig Filterable f => (a -> Boolean) -> f a -> f a
+ * @sig (a -> Boolean) -> [a] -> [a]
+ * @sig (a -> Boolean) -> StrMap a -> StrMap a
+ * @sig (Applicative f, Foldable f, Monoid (f a)) => (a -> Boolean) -> f a -> f a
+ * @sig (Alternative m, Monad m) => (a -> Boolean) -> m a -> m a
```

Comment:
Although I'm greatly in favor of expanding our type signatures in this way, and I'm not particularly worried about scaring people with all the types (except maybe `StrMap`: "What happened to my `Object`?!" :smile:), I am worried about losing the description of "unprincipled dispatching" in `Filterable f => (a -> Boolean) -> f a -> f a`. where `Filterable` would point to a description that specifies only having for Type T, `filter:: t a ~> (a -> Boolean) -> t a`.

Reply:
I don't think we should use type classes for unprincipled dispatching. It's okay† to decide to dispatch to a `quux` method, and we should document this if we do so, but I'd oppose inventing a lawless type class named Quuxable and using `Quuxable q =>` in type signatures.

If you'd like a lawful Filterable type class so we could remove the long-winded signatures of the derivations, please support fantasyland/fantasy-land#274. :)

---

† _As I've stated, I'm opposed to dispatching to methods which don't claim to conform to a specification, but I'll set aside this opposition for this discussion._

Reply:
> I don't think we should use type classes for unprincipled dispatching.

Yes, I think you're right.  We do need to document the names of methods to which we might dispatch.

Although I have not followed FL all that much lately, I am watching that issue; I just haven't had anything to contribute yet.